services/datamanager/builtin/capture_test.go

```diff
@@ -341,18 +342,48 @@ func testFilesContainSensorData(t *testing.T, dir string) {
 	}
 }
 
-// waitForCaptureFilesToExceedNFiles returns once `captureDir` contains more than `n` files.
+// waitForCaptureFilesToExceedNFiles returns once `captureDir` contains more than `n` files of at
+// least `emptyFileBytesSize` bytes. This is not suitable for waiting for file deletion to happen.
 func waitForCaptureFilesToExceedNFiles(captureDir string, n int) {
-	totalWait := time.Second * 2
-	waitPerCheck := time.Millisecond * 10
-	iterations := int(totalWait / waitPerCheck)
-	files := getAllFileInfos(captureDir)
-	for i := 0; i < iterations; i++ {
-		if len(files) > n && files[n].Size() > int64(emptyFileBytesSize) {
+	for {
+		files := getAllFileInfos(captureDir)
+		nonEmptyFiles := 0
+		for idx := range files {
+			if files[idx].Size() > int64(emptyFileBytesSize) {
+				// Every datamanager file has at least 90 bytes of metadata. Wait for that to be
+				// observed before considering the file as "existing".
+				nonEmptyFiles++
+			}
+
+			// We have N+1 files. No need to count any more.
+			if nonEmptyFiles > n {
```

Comment:
TestFilePolling wanted "wait for exactly" which this code doesn't do.

But the old version of this code had a 2 second timeout so this function looked like it was doing what was needed.